### PR TITLE
fix(terraform): Outdated check for google_container_cluster binary authorization

### LINKED ISSUE #3409: Outdated check for `google_container_cluster` binary authorization
**Describe the issue**
`Check: CKV_GCP_66: "Ensure use of Binary Authorization"` is not checking [against the updated property](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#nested_binary_authorization) causing the check to fail.

The current property states that the old method [`enable_binary_authorization`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#enable_binary_authorization) is deprecated in favor the new [`binary_authorization`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#nested_binary_authorization) block

**Examples**

Before:
```hcl
enable_binary_authorization = true
```

Now:
```hcl
  binary_authorization {
    evaluation_mode = "PROJECT_SINGLETON_POLICY_ENFORCE"
  }
```

**Version (please complete the following information):**
 - Checkov Version 2.1.149

### PR DESCRIPTION
**Pull request by losisin**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

This PR tries to cover all cases how binary authorization can be enabled for `google_container_cluster`  in different version of google provider plugin. Hopefully .... :)

Fixes #3409 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/gcp/GKEBinaryAuthorization.py [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+            binary_authorization = conf["binary_authorization"][0]
+            if isinstance(binary_authorization, dict) and 'evaluation_mode' in binary_authorization:
+                # Google provider version >= v4.31.0
+                if binary_authorization.get("evaluation_mode") == ["PROJECT_SINGLETON_POLICY_ENFORCE"]:
+                    return CheckResult.PASSED
+                # Google provider version v4.29.0 and v4.30.0
+                elif binary_authorization.get("evaluation_mode") == [True]:
+                    return CheckResult.PASSED
+        # Google provider version <= v4.28.0
+        if conf.get("enable_binary_authorization") and conf.get("enable_binary_authorization") == [True]:
```
Comment (gruebel) [inline checkov/terraform/checks/resource/gcp/GKEBinaryAuthorization.py:24]:
```suggestion
        if conf.get("enable_binary_authorization") == [True]:
```
no need to check for being not empty and then actually checking it is equal 🙂

**Review by gruebel — COMMENTED** [review]
nice good job, just a a small suggestion

File: checkov/terraform/checks/resource/gcp/GKEBinaryAuthorization.py (hunk 1 above)
Comment (losisin) [reply checkov/terraform/checks/resource/gcp/GKEBinaryAuthorization.py]: done. thanks!

**Review by gruebel — APPROVED** [review]
thanks, looks great 🍺

**Comment by losisin** [comment]
@gruebel do I need to reach out to someone else for another approve review?

**Comment by gruebel** [comment]
hey @losisin sadly you need to wait a bit, because my colleagues have now a couple of holidays 🏖️ but when they are back, I will grab some to review your PR 💪

**Review by tronxd — APPROVED** [review]

**Comment by gruebel** [comment]
@losisin tried to fix the merge conflict, but I have no permission to push my change to your branch

**Comment by losisin** [comment]
@gruebel I resolved the conflict and did `pre-commit run --all-files` just in case. I think we are good to go

**Pull request merged**